README.md

```diff
@@ -94,6 +96,10 @@ The `NB_API_ALLOWED_ORIGINS` variable defaults to an empty string (`""`) when un
 However, in many cases you may want to make the API accessible by a frontend tool such as our [browser query tool](https://github.com/neurobagel/query-tool).
 To do so, you must explicitly specify the origin(s) for the frontend using `NB_API_ALLOWED_ORIGINS` in `.env`. 
 
+When you [locally deploy the API with `docker compose`](#docker), you may also deploy a local version of the 
```

Comment:
General comment:

Reading this PR, I see what you were saying the other day about the complexity of having the "setup" for multiple tools in the same repo (i.e., the API repo)...

Since a locally hosted query tool is not required to use a local API and graph and doesn't need to be in the same network as the other two (whereas the API and graph should generally be spun up together, and wouldn't really be used separately in our current setup IMO), I think having the query tool in the `docker-compose.yml` is really just a convenience feature for users. That said, I think we should still provide + emphasize the option to spin up _just_ the API and graph using `docker-compose` (e.g., if the deploying group wants to/if we want to encourage them to make their data accessible using our public query tool instead of their own).

One way to do this could be to have **two** Compose files here, one with our old API + graph config, and one that extends that base file with an extra service for the query tool (see https://docs.docker.com/compose/multiple-compose-files/extends/). Then, we could instruct the user to use one or the other based on what tools they need (and they always have the option to deploy the query tool later using the instructions from https://github.com/neurobagel/query-tool).

Some more meta comments on this:
I'm realizing that other than the convenience of running just one `docker compose` command instead of two (`docker compose` + `docker run`), I don't see a _super_ compelling reason to have the query tool in the same Compose config at all, or at least not as part of our "recommended" setup route (see two-Compose-file possibility?) 🤡. Not sure if we jumped the gun a little, but IIRC our main goal was to have a query tool that could be spun up locally for sysadmins, which is now already possible since it's Dockerized. Also, to my understanding we eventually still want to achieve a federated-search-from-a-single-public-query-tool situation, right? Rather than a collection of unconnected institutes that can only search internally (or at least, I imagine this won't be our primary focus?). I wonder then whether our guidelines here will send a conflicting message - maybe we want to instead turn this into a documentation problem, where we continue to suggest Compose for deploying the API and graph, but have separate recommendations for a specific restricted-clinical-data use case.

Sorry for the word dump! Interested to hear your thoughts on this, or to continue this discussion offline.

Reply:
I generally agree.

> I think having the query tool in the docker-compose.yml is really just a convenience feature for users

yes. But for a deploying sysAdmin, convenience **is** an important feature. So in that situation we want a single `docker-compose.yml` for all three services.

> I think we should still provide + emphasize the option to spin up just the API and graph using docker-compose

Also agree, but this is for a different situation (e.g. local dev, running tests, or deploying to prod in a world where the query tool is hosted elsewhere). 

> One way to do this could be to have two Compose files here

Again, full agree. I think we should have 

- API+Graph `docker-compose.yml` inside the **API** repo. Then the file is used for local dev, tests, etc.
- API+Graph+query tool `docker-compose.yml` inside a -to-be-created-location for **setup and installation** instructions. This could be easily copy-pasteable code blocks in our docs, or even a dedicated repo for these files, or a zip file. The target use would be: "I want to deploy a fully isolated, local deployment". And we already have two places that need this.

> to my understanding we eventually still want to achieve a federated-search-from-a-single-public-query-tool situation,

yes on the `federated-search`, maybe on the `from-a-single-public-query-tool`. I'm not sure about the second part because I'm not sure we will say "any federated query has to go through the same piece of Javascript that lives at this URL". I do agree that we want to be clear in our communication about this though.

Reply:
Good points. We can continue the discussion about multiple Compose files here https://github.com/neurobagel/api/issues/162